scripts/behavior.js

```diff
@@ -381,6 +463,123 @@ function executeCommand(command, arguments, cooldown, message, settings, utils,
 ///
 ///////////////
 
+/**
+ * Checks wether a role is already present
+ * @param {String} roleName The role to check
+ * @param {Guild} guild The guild to investigate
+ * @returns {boolean} Wether this role is already present in the guild specifics
+ */
+function isRoleAlreadyExisting(roleName, guild) {
+    let specifics = utils.getSpecifics(guild);
+    return specifics["registeredRoles"].some(role => role === roleName)
+}
+
+/**
+ * Creates a new role in the specifics and on the discord server
+ * Won't check if the role is present
+ * @param {String} roleName The name of the role to create
+ * @param {Message} message The message from which to parse guild and author
+ * @returns {Promise<any>} A promise that completes successfully when the role creation succeeded and completes exceptionally when the creation failed due to the discord server
+ */
+function createRole(roleName, message) {
+    return new Promise((resolve, reject) => {
+        let specifics = utils.getSpecifics(message.guild);
+        specifics["registeredRoles"].push(roleName);
+        utils.writeSpecifics(message.guild, specifics);
+
+        message.guild.createRole({name: arguments, color: "LIGHT_GREY", mentionable: true, hoist: false}, "Created by Dostya as requested by" + message.author.username)
+            .then(value => {
+                utils.log("Created role ${roleName} as requested by discord user ${message.author.id}", "--", message.guild);
+                replyToMessage(message, "Role created")
+                    .then(resolve());
+            }, rejectReason => {
+                let specificsCreateRoleFailed = utils.getSpecifics(message.guild);
+                specificsCreateRoleFailed.registeredRoles = specificsCreateRoleFailed.registeredRoles.filter(role => role !== roleName);
+                utils.writeSpecifics(message.guild, specificsCreateRoleFailed);
+                utils.log("Error while creating role. Reason: " + rejectReason, "--", message.guild);
+                replyToMessage(message, "Could not create role due to internal discord error. Please contact an administrator. Reason: " + rejectReason)
+                    .then(reject());
+            });
+    });
+}
+
+/**
+ * Deletes a role in the specifics and on the discord server
+ * Won't check if the role is present
+ * @param {String} roleName The name of the role to delete
+ * @param {Message} message The message from which to parse guild and author
+ * @returns {Promise<any>} A promise that completes successfully when the role deletion succeeded and completes exceptionally when the deletion failed due to the discord server
+ */
+function deleteRole(roleName, message) {

```

Comment:
If deleterole returns a promise it should be async right ?

Reply:
It already is async due to the promise, but yes,  I will restructure createrole and deleterole to use the "new" async keyword, easier to read.